CMake Error at libobs/cmake/os-windows.cmake:46 (target_link_libraries):   Target "libobs" links to:      OBS::w32-pthreads

### Operating System Info

Windows 11

### Other OS

_No response_

### OBS Studio Version

Git

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

None

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

No UI build 

### Current Behavior

CMake Error at libobs/cmake/os-windows.cmake:46 (target_link_libraries):
  Target "libobs" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  libobs/CMakeLists.txt:251 (include)


CMake Error at plugins/image-source/CMakeLists.txt:13 (target_link_libraries):
  Target "image-source" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/obs-ffmpeg/CMakeLists.txt:99 (target_link_libraries):
  Target "obs-ffmpeg" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/obs-ffmpeg/ffmpeg-mux/CMakeLists.txt:14 (target_link_libraries):
  Target "obs-ffmpeg-mux" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/obs-filters/CMakeLists.txt:39 (target_link_libraries):
  Target "obs-filters" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/obs-outputs/CMakeLists.txt:56 (target_link_libraries):
  Target "obs-outputs" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at deps/happy-eyeballs/CMakeLists.txt:13 (target_link_libraries):
  The link interface of target "happy-eyeballs" contains:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at deps/file-updater/CMakeLists.txt:10 (target_link_libraries):
  The link interface of target "file-updater" contains:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/vlc-video/CMakeLists.txt:39 (target_link_libraries):
  Target "vlc-video" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/win-capture/CMakeLists.txt:68 (target_link_libraries):
  Target "win-capture" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



CMake Error at plugins/win-dshow/CMakeLists.txt:28 (target_link_libraries):
  Target "win-dshow" links to:

    OBS::w32-pthreads

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (1.7s)

### Steps to Reproduce

cmake --preset windows-x64 -DENABLE_UI:BOOL=OFF

### Anything else we should know?

_No response_

--- a/libobs/cmake/os-windows.cmake
+++ b/libobs/cmake/os-windows.cmake
@@ -1,3 +1,7 @@
+if(NOT TARGET OBS::w32-pthreads)
+  add_subdirectory("${CMAKE_SOURCE_DIR}/deps/w32-pthreads" "${CMAKE_BINARY_DIR}/deps/w32-pthreads")
+endif()
+
 configure_file(cmake/windows/obs-module.rc.in libobs.rc)
 
 add_library(obs-obfuscate INTERFACE)

